crates/sui-swarm-config/src/node_config_builder.rs

```diff
@@ -135,6 +137,39 @@ impl ValidatorConfigBuilder {
         }
     }
 
+    fn build_narwhal_config(
+        for_testing: bool,
+        narwhal_metrics_address: Multiaddr,
+    ) -> narwhal_config::Parameters {
+        let localhost = local_ip_utils::localhost_for_testing();
+        let network_admin_server = NetworkAdminServerParameters {
+            primary_network_admin_server_port: local_ip_utils::get_available_port(&localhost),
+            worker_network_admin_server_base_port: local_ip_utils::get_available_port(&localhost),
+        };
+        let prometheus_metrics = PrometheusMetricsParameters {
+            socket_addr: narwhal_metrics_address,
+        };
+        if for_testing {
+            narwhal_config::Parameters {
+                network_admin_server,
+                prometheus_metrics,
+                // NOTE: the following parameters are important to ensure tests run fast. Using the default
+                // Narwhal parameters may result in tests taking >60 seconds.
```

Comment:
does this actually produce a significant difference when using simtest? if not, we may want to only use this change for tokio tests, so that simtest can stay as close as possible to the production config.